spec.emu

```diff
@@ -882,6 +882,297 @@ contributors: Google, Ecma International
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendarmonthsinyear" type="implementation-defined abstract operation">
+      <h1>
+        CalendarMonthsInYear (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ as arithmetical value in the given _calendar_ and returns the number of months in the corresponding year.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-regulatenonisoyearmonth" type="implementation-defined abstract operation">
+      <h1>
+        RegulateNonISOYearMonth (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _arithmeticalYear_: an integer,
+          _monthCode_: a String,
+          _years_: an integer,
+        ): a Record with fields [[Year]] (an integer) and [[Month]] (an integer)
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          Interprets _isoDate_ as a date in _calendar_, adds _years_, and returns the best-match arithmetic _year_ and _month_ according to the calendar.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _y2_ be _arithmeticalYear_ + _years_.
+        1. If _calendar_ is *"chinese"*, *"dangi"*, or *"hebrew"*, then
+          1. If _calendar_ is *"chinese"* or *"dangi"*, then
+            1. Let _month_ be MonthCodeToArithmeticalMonth(_calendar_, _y2_, _monthCode_).
+            1. If _month_ is ~invalid~, then
+              1. Let _monthCode_ be CreateMonthCode(ParseMonthCode(_monthCode_).[[Number]], *false*).
+              1. Set _month_ to MonthCodeToArithmeticalMonth(_calendar_, _y2_, _monthCode_).
+              1. Assert: _month_ is not ~invalid~.
+          1. Else,
+            1. Assert: _calendar_ is *"hebrew"*.
+            1. Let _isLeap1_ be *true* if _arithmeticalYear_ is a leap year in the Hebrew calendar, or *false* otherwise.
+            1. Let _isLeap2_ be *true* if _y2_ is a leap year in the Hebrew calendar, or *false* otherwise.
+            1. If _isLeap1_ is *true* and _isLeap2_ is *false*, then
+              1. If _monthCode_ is *"M05L"*, then
+                1. Set _monthCode_ to *"M06"*.
+            1. Else if _isLeap1_ is *false* and _isLeap2_ is *true*, then
+              1. If _monthCode_ is *"M06"*, then
+                1. Set _monthCode_ to *"M05L"*.
```

Comment:
interesting choice; any particular reason behind it?

Not that I dislike it, but I would have thought keeping M06 as M06 would be clearer since Adar II is supposed to be the "regular" Adar

Reply:
https://github.com/tc39/proposal-intl-era-monthcode/issues/70

Reply:
I deleted this AO in 6abbe6e3ca1ca220e33f3a62cb474fe631a75495 and replaced it with a table reference